### ISSUE TITLE
[OSX] nimble install gintro fails - unexpected gc

### ISSUE THREAD

**Issue by jrods** [StefanSalewski/gintro#66]
I'm currently interested in trying to use nim & gtk3 on Mac 10.15 (first time), but it fails to install via error below (my skills and knowledge with nim are beginner level). From what I've research seems to be related to the gc? Is there a flag I should be including when running nimble install? Or is it relating to my version of him that I've installed?

Nim version: 1.0.6 (installed with brew, both bottle and build from source fail with same results)
Mac OS X: 10.15.3

```
$ nimble install gintro --verbose

Hint: combinatorics [Processing]
/private/var/folders/fn/t2yhy2c935ddnl55zl6zdyfm0000gn/T/gintrosalewski/gen.nim(2519, 21) Error: 'none', 'boehm' or 'refc' expected, but 'arc' found
stack trace: (most recent call last)
/private/var/folders/fn/t2yhy2c935ddnl55zl6zdyfm0000gn/T/nimblecache/nimscriptapi.nim(177, 29)
/private/var/folders/fn/t2yhy2c935ddnl55zl6zdyfm0000gn/T/nimble_53234/githubcom_stefansalewskigintro/gintro_53234.nims(81, 7) installBefore
/private/var/folders/fn/t2yhy2c935ddnl55zl6zdyfm0000gn/T/nimble_53234/githubcom_stefansalewskigintro/gintro_53234.nims(66, 7) prep
/usr/local/Cellar/nim/1.0.6/nim/lib/system/nimscript.nim(252, 7) exec
/usr/local/Cellar/nim/1.0.6/nim/lib/system/nimscript.nim(252, 7) Error: unhandled exception: FAILED: nim c gen.nim [OSError]
     Error: Exception raised during nimble script execution
```

**Comment by StefanSalewski** [comment]
Thanks for reporting.

That is an interesting error, maybe related to latest update to v0.7.1 some days ago. Will investigate it tomorrow.

But note, most people say that GTK on Mac is not much fun. (Not only for Nim, but generally.) You may see also https://github.com/nim-lang/Nim/wiki/Curated-Packages#gui

**Comment by jrods** [comment]
Thanks for the quick response, if you need anymore info or suggestions on trying things, I'll be watching this issue.

I noticed that libgc wasn't install, so I tried installing libgc and did a recompile of nim, but same results. I'm not sure if there's more to it than that.

When I have time to look at this again, I could try an older release of gintro.

_I'm interested in gtk mainly for the possible system resource lightweight-ness of it, the desktop consistency (on the linux side anyways) and learning something new. My second choice would be nim's webview._

**Comment by StefanSalewski** [comment]
I think I already know the issue:

https://github.com/StefanSalewski/gintro/blob/master/tests/gen.nim#L2518

when compileOption("gc", "arc"):

I have that statement in many places. Thought that is a string test, but maybe an enum test, and arc enum is only available in devel compiler?

Have to ask Araq, will fix it tomorrow. You may consider installing Nim devel if you are not using Nim in production, I use Nim devel on linux all the time, works fine. Many people use devel. Or you may install gintro < 0.7, I think that was 0.6.4. But maybe better wait for my fix.

**Comment by StefanSalewski** [comment]
OK, I have tried to fix this issue.

Problem was indeed that statement compileOption("gc", "arc") does not compile for Nim 1.0.x. So I have substituted it with defined(gcDestructors) as suggested in

https://forum.nim-lang.org/t/5965

Araq or another core dev was not available today, as we have sunday here in europe, but I tested in Nim playground that defined(gcDestructors) compiles with Nim 1.0.6. And for me it seems still to work, so I hope all is fine now.

For MacOS -- well there may be other problems as well, some people do not manage to install GTK or a working C compiler on Windows or Mac at all...

Let me know if there are more problems.
